server/datastore/mysql/migrations/tables/20240424124712_AddSoftwareInstallerTables.go

```diff
@@ -0,0 +1,117 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240424124712, Down_20240424124712)
+}
+
+func Up_20240424124712(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE IF NOT EXISTS software_installers (
+  id int(10) unsigned NOT NULL AUTO_INCREMENT,
+
+  -- FK to the "software version" this installer matches
+  software_id bigint(20) unsigned DEFAULT NULL,
+
+  -- Raw osquery SQL statment to be run as a pre-install condition
+  pre_install_query text COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+
+  -- FK to the script_contents for the script used to install this software
+  install_script_content_id int(10) unsigned NOT NULL,
+
+  -- FK to the script_contents for the post-script uploaded by the IT admin to
+  -- be run after the software is installed 
+  post_install_script_content_id int(10) unsigned DEFAULT NULL,
+
+  -- used to track the ID retrieved from the storage containing the installer bytes
+  storage_id binary(16) NOT NULL,
```

Comment:
Might want to expand that to 64 bytes so we can use sha256 (the binary won't go through mysql so we shouldn't limit ourselves to md5 more than needed).